flake.nix

```diff
@@ -0,0 +1,184 @@
+    Go libary for building CSI drivers to deliver cert-manager signed
+    certificate key pairs to Kubernetes workloads.
+  '';
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { inherit system; overlays = [ (import ./hack/nix/overlay-kind.nix) ]; };
+
+        # We only source go files to have better cache hits when actively
+        # working on non-go files.
+        src = pkgs.lib.sourceFilesBySuffices ./. [ ".go" "go.mod" "go.sum" ];
+        vendorSha256 = "sha256-3pNKmR+yKIf/15eftJyHD7m7LerFbZ2m+N6zxVXz2sU=";
+
+
+        e2e-cert-manager-version = "1.9.1";
+        e2e-cert-manager-controller-digest = "sha256:cd9bf3d48b6b8402a2a8b11953f9dc0275ba4beec14da47e31823a0515cde7e2";
+        e2e-cert-manager-controller-sha256 = "sha256-NQcTUOuqmHDWqD8kMhE8AApZmsNa3ElXlHe5qyCrSJs=";
+        e2e-cert-manager-webhook-digest = "sha256:4ab2982a220e1c719473d52d8463508422ab26e92664732bfc4d96b538af6b8a";
+        e2e-cert-manager-webhook-sha256 = "sha256-Nr1xSnXhIdobf3vPqnf8iS/8FbXN+yEEz+KpdmVKB3w=";
+        e2e-cert-manager-cainjector-digest = "sha256:df7f0b5186ddb84eccb383ed4b10ec8b8e2a52e0e599ec51f98086af5f4b4938";
+        e2e-cert-manager-cainjector-sha256 = "sha256-Bf9AJzjwnsF3oi37UAMa9DN9LXZ2AMlt9kjpsRyFBHg=";
+        e2e-cert-manager-ctl-digest = "sha256:468c868b2cbae19a5d54d34b6f1c27fe54b0b3988a6d8cab74455f5411d95e96";
+        e2e-cert-manager-ctl-sha256 = "sha256-D0ZOewtFI3on6U2ALdbBZZwlwA6uPkyQLXnCbMlZyoQ=";
+
+        e2e-kind-node-version = "1.25.0";
+        e2e-kind-node-digest = "sha256:428aaa17ec82ccde0131cb2d1ca6547d13cf5fdabcc0bbecf749baa935387cbf";
+        e2e-kind-node-sha256 = "sha256-s8kIavYmei38a+PrWjj54BDW9grLP8tRU09XkQ2zAME=";
+
+        csi-node-driver-registrar-version = "1.2.0";
+        csi-node-driver-registrar-digest = "sha256:89cdb2a20bdec89b75e2fbd82a67567ea90b719524990e772f2704b19757188d";
+        csi-node-driver-registrar-sha256 = "sha256-yve52Xxoyc00lOJGPACijt+3VQX6ngXtfl6nGqbFsIw=";
+
+        e2e-busybox-digest = "sha256:b8f68c62fe862281bf598060f15cb080ef778dc9db19f136d19a3531ffcb9aa0";
+        e2e-busybox-sha256 = "sha256-E0PakwKrCGFybyKC3BW/LW43oHiiCXQuT4dKtHrdNc4=";
+
+        cert-manager-csi = pkgs.buildGo119Module {
+          name = "cert-manager-csi";
+          inherit src vendorSha256;
+          subPackages = [ "./example" ];
+          postInstall = "mv $out/bin/example $out/bin/cert-manager-csi";
+        };
+
+        # e2e test binary.
+        csi-lib-e2e = pkgs.buildGo119Module {
+          name = "csi-lib-e2e";
+          inherit src vendorSha256;
+          buildPhase = ''
+            go test -v --race -o csi-lib-e2e -c ./test/e2e/.
+          '';
+          postInstall = ''
+            mkdir -p $out/bin
+            mv csi-lib-e2e $out/bin/.
+          '';
+          doCheck = false;
```

Comment:
What is `doCheck` and why is it disabled here but not mentioned in the `cert-manager-csi` definition?

Reply:
Not sure why `doCheck` was included. Think that was a copy+paste error.